### ISSUE TITLE
upload_to do not take a function as arg

### ISSUE THREAD

**Issue by spudnica** [edoburu/django-private-storage#3]
Hi!

I just tested this app and it works perfectly but I cant set user upload folders to their username. On default django FileField I used custom function which returns path with username and date:

 

```
def func(instance, filename):
    time = datetime.now()
    return '%s/%s/%s' % (instance.user.username, time.year, filename)
```
`file = PrivateFileField(upload_to=func)`
```

With PrivateFileField above returns and AttributeError: 'function' object has no attribute 'endswith'

Traceback:

File "/some/path/to/proj-env/local/lib/python2.7/site-packages/django/core/handlers/exception.py" in inner
  39.             response = get_response(request)

File "/some/path/to/proj-env/local/lib/python2.7/site-packages/django/core/handlers/base.py" in _get_response
  187.                 response = self.process_exception_by_middleware(e, request)

File "/some/path/to/proj-env/local/lib/python2.7/site-packages/django/core/handlers/base.py" in _get_response
  185.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/some/path/to/proj-env/local/lib/python2.7/site-packages/django/views/generic/base.py" in view
  68.             return self.dispatch(request, *args, **kwargs)

File "/some/path/to/proj-env/local/lib/python2.7/site-packages/django/views/generic/base.py" in dispatch
  88.         return handler(request, *args, **kwargs)

File "/some/path/to/proj/szcz/views.py" in post
  76. 				PL = Dszcz.objects.create(plik2=myfile, user=request.user, postepowanie_id=int(self.kwargs['postepowanie_id']))

File "/some/path/to/proj-env/local/lib/python2.7/site-packages/django/db/models/manager.py" in manager_method
  85.                 return getattr(self.get_queryset(), name)(*args, **kwargs)

File "/some/path/to/proj-env/local/lib/python2.7/site-packages/django/db/models/query.py" in create
  399.         obj.save(force_insert=True, using=self.db)

File "/some/path/to/proj-env/local/lib/python2.7/site-packages/django/db/models/base.py" in save
  796.                        force_update=force_update, update_fields=update_fields)

File "/some/path/to/proj-env/local/lib/python2.7/site-packages/django/db/models/base.py" in save_base
  824.             updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)

File "/some/path/to/proj-env/local/lib/python2.7/site-packages/django/db/models/base.py" in _save_table
  908.             result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)

File "/some/path/to/proj-env/local/lib/python2.7/site-packages/django/db/models/base.py" in _do_insert
  947.                                using=using, raw=raw)

File "/some/path/to/proj-env/local/lib/python2.7/site-packages/django/db/models/manager.py" in manager_method
  85.                 return getattr(self.get_queryset(), name)(*args, **kwargs)

File "/some/path/to/proj-env/local/lib/python2.7/site-packages/django/db/models/query.py" in _insert
  1045.         return query.get_compiler(using=using).execute_sql(return_id)

File "/some/path/to/proj-env/local/lib/python2.7/site-packages/django/db/models/sql/compiler.py" in execute_sql
  1053.             for sql, params in self.as_sql():

File "/some/path/to/proj-env/local/lib/python2.7/site-packages/django/db/models/sql/compiler.py" in as_sql
  1006.                 for obj in self.query.objs

File "/some/path/to/proj-env/local/lib/python2.7/site-packages/django/db/models/sql/compiler.py" in pre_save_val
  955.         return field.pre_save(obj, add=True)

File "/some/path/to/proj-env/local/lib/python2.7/site-packages/django/db/models/fields/files.py" in pre_save
  292.             file.save(file.name, file, save=False)

File "/some/path/to/proj-env/local/lib/python2.7/site-packages/django/db/models/fields/files.py" in save
  90.         name = self.field.generate_filename(self.instance, name)

File "/some/path/to/proj-env/local/lib/python2.7/site-packages/private_storage/fields.py" in generate_filename
  74.         return smart_str(os.path.join(*dirs))

File "/some/path/to/proj-env/lib/python2.7/posixpath.py" in join
  77.         elif path == '' or path.endswith('/'):
```

**Comment by AndyLPK247** [comment]
Hi @spudnica ,

Based on the README doc and the source code in **private_storage.fields**, it looks like the functionality of _upload_to_ is meant to be different for _PrivateFileField_ than for the standard _FileField_. The _upload_to_ argument should be a string. The _upload_subfolder_ argument should take a function. It also looks like the function is different - it takes only the instance but not the filename, and it returns only the subfolder path instead of the full file path. Please try using _upload_subfolder_.

**Comment by vdboor** [comment]
Thanks @AndyLPK247 for the detailed explaination. That is indeed correct :-)

**Comment by vdboor** [comment]
Both formats are now supported in master. Could you please test this latest version? (and tell me which Django version you've tested against?)

**Comment by vdboor** [comment]
Update, after writing unit tests I've released v2.0 :-)
